cosmwasm/utils.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const { readFileSync } = require('fs');
+const { calculateFee, GasPrice } = require('@cosmjs/stargate');
+const { instantiate2Address } = require('@cosmjs/cosmwasm-stargate');
+const { getSaltFromKey } = require('../evm/utils');
+
+const pascalToSnake = (str) => {
+    return str.replace(/([A-Z])/g, (group) => `_${group.toLowerCase()}`).replace(/^_/, '');
+};
+
+const uploadContract = async (config, options, wallet, client) => {
```

Comment:
```suggestion
const uploadContract = async (config, { artifactPath, contractName, instantiate2, salt, contractName }, wallet, client) => {
```
This would work, too. I'll leave it to you.